### PR TITLE
Set up CI to run buck2 samples

### PR DESCRIPTION
**Pull request by nlopezgi**
- Creating PR to run Buck2 samples in CI
- Uses github auth to authenticate buck2 client with EngFlow cluster - https://docs.engflow.com/re/config/options.html#experimental_github_auth_container
- Uses latest version of Buck2, if we start having issues with this approach will migrate to use of dotslash for configs.
- Runs the cpp, golang and python Buck2 samples, as more are added they need to be include in the test script
- DONE: use pre-installed zstd once I release images with the package

### CODE REVIEW

File: infra/test-buck2.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+set -xe
+
+# Run cpp example
+cd buck2/cpp
+../../buck2-exe build //:cpp_lib
```
Comment (TheGrizzlyDev) [inline infra/test-buck2.sh:6]: can't you create an alias instead and use an absolute path? Something like `alias buck2="$(realpath buck2-exe)"` and then just invoke buck2 normally

**Review by TheGrizzlyDev — APPROVED** [review]
Nit

File: infra/test-buck2.sh (hunk 1 above)
Comment (nlopezgi) [reply infra/test-buck2.sh]: done

File: infra/test-buck2.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+set -xe
+shopt -s expand_aliases
```
Comment (TheGrizzlyDev) [inline infra/test-buck2.sh:3]: You could have also used a function FYI

**Review by TheGrizzlyDev — APPROVED** [review]

File: infra/test-buck2.sh (hunk 2 above)
Comment (nlopezgi) [reply infra/test-buck2.sh:3]: Thanks. I will consider that option in the general case. In this case I think using a function might be too much, but if the script starts getting more complex as we add more samples I will switch to using a function.

**Pull request merged**